provider/pkg/provider/helm_release.go

```diff
@@ -819,34 +741,34 @@ func (r *helmReleaseProvider) Diff(ctx context.Context, req *pulumirpc.DiffReque
 	logger.V(9).Infof("Diff: Old release: %#v", oldRelease)
 	logger.V(9).Infof("Diff: New release: %#v", newRelease)
 
-	// Always set desired state to DEPLOYED
-	// TODO: This could be done in Check instead?
-	if newRelease.Status == nil {
-		newRelease.Status = &ReleaseStatus{}
-	}
-	newRelease.Status.Status = release.StatusDeployed.String()
-
-	oldInputsJSON, err := json.Marshal(oldInputs.Mappable())
+	// Generate a patch to apply the new inputs to the old state, including deletions.
+	schema := &noSchema{}
+	oldInputsJSON, err := json.Marshal(oldInputs.MapRepl(nil, mapReplStripSecrets))
 	if err != nil {
-		return nil, err
+		return nil, pkgerrors.Wrapf(err, "internal error: json.Marshal(oldInputsJson)")
 	}
-	newInputsJSON, err := json.Marshal(news.Mappable())
+	logger.V(9).Infof("oldInputsJSON: %+v", string(oldInputsJSON))
+	newInputsJSON, err := json.Marshal(news.MapRepl(nil, mapReplStripSecrets))
 	if err != nil {
-		return nil, err
+		return nil, pkgerrors.Wrapf(err, "internal error: json.Marshal(oldInputsJson)")
 	}
-
-	logger.V(9).Infof("oldInputsJSON: %+v", string(oldInputsJSON))
 	logger.V(9).Infof("newInputsJSON: %+v", string(newInputsJSON))
-	patch, err := jsonpatch.CreateMergePatch(oldInputsJSON, newInputsJSON)
+	oldStateJSON, err := json.Marshal(olds.MapRepl(nil, mapReplStripSecrets))
 	if err != nil {
-		return nil, err
+		return nil, pkgerrors.Wrapf(err, "internal error: json.Marshal(oldStateJson)")
+	}
+	logger.V(9).Infof("oldStateJson: %+v", string(oldStateJSON))
+	strategicPatch, err := strategicpatch.CreateThreeWayMergePatch(oldInputsJSON, newInputsJSON, oldStateJSON, schema, true)
+	if err != nil {
+		return nil, pkgerrors.Wrapf(err, "internal error: CreateThreeWayMergePatch")
 	}
+	logger.V(1).Infof("strategicPatch: %+v", string(strategicPatch))
```

Comment:
Let's update it to `%s` since it's a string output. I believe linters would catch it if we have a type mismatch incase we accidentally drop the `string()` process in the future.